packages/terra-table/src/SelectableTableRows.jsx

```diff
@@ -62,12 +116,41 @@ class SelectableTableRows extends React.Component {
 
   wrappedOnKeyDownForRow(row, index) {
     const initialOnKeyDown = row.props.onKeyDown;
-
     return (event) => {
       if (event.nativeEvent.keyCode === KeyCode.KEY_RETURN || event.nativeEvent.keyCode === KeyCode.KEY_SPACE) {
         // The default isSelectable attribute is either undefined or true, unless the consumer specifies the row's isSelectable attribute as false.
         if (row.props.isSelectable !== false) {
           this.handleOnChange(event, index);
+
+          const {
+            liveRegion,
+            screenReaderUpdateOnSelected,
+            selectedIndexes,
+          } = this.props;
+
+          const { intl } = this.context;
+
+          if (screenReaderUpdateOnSelected) {
+            screenReaderUpdateOnSelected(liveRegion, intl);
+          } else {
+            const isSelected = selectedIndexes.indexOf(index) >= 0;
```

Comment:
these will make VO to read unselected row when we click on preselected rows like shown in this : [example](https://terra-core-deployed-pr-2331.herokuapp.com/#/tests/terra-table/table/selectable-table/selectable-table-rows-preselected) 

In the above example table has row [0] and row [2] in selectedIndexes which makes this condition ``` selectedIndexes.indexOf(index) >= 0 ``` to return true on click of row and adds row unselected text to live region even when the row is still selected. 

This works fine when rows are pre-selected using isSelected prop where we can de-select the pre-selected row on click, but Pre-selected rows specified using selectedIndexes cannot be de-selected on click, this will always keep the row in selected state. 

for this scenario we might need to keep a flag which will be true when `this.props.selectedIndexes` has values initially and false when `this.props.selectedIndexes` is not initialized. 

By Adding this flag to below condition we can prevent VO from reading pre-selected rows as unselected rows
`if(!isSelected && isSelectedIndexesIntialized)`

Reply:
selectedIndexes will actually always be initialized to an empty array. I did try testing it with an empty array, and the screen reader wasn't able to correctly specify whether the row was selected or deselected. For this case, maybe we can tie onto the isSelected prop if that is what is mean to always keep a row selected?

Reply:
I did an investigation. This is not a perfect solution, and it requires the end user insuring they coded their onChange correctly. If you want the correct selected status to be read when a row is clicked, whether it's preselected or not, you can check for the presence of an onChange. If that does not exist, then we can check the value of the isSelected prop on the row itself.

It's not ideal, but I haven't found other ways that let us determine if the clicked row is selected or not. This component has a lot of moving parts, and it's getting hard to determine the point of change from when a row goes from selected to unselected.